usr/bin/cloudguestregistryauth

```diff
@@ -20,12 +20,21 @@ feature.
 The script is used in a sudo rule to allow its use by a non root user
 """
 
+import os
 import sys
 
-from cloudregister.registerutils import get_activations
-
-
-if not get_activations():
-    sys.exit('Could not refresh credentials')
+from cloudregister.registerutils import (
+    get_activations, exec_subprocess
+)
+
+error_message = 'Could not refresh credentials'
+if os.geteuid():
+    if exec_subprocess(['sudo', sys.argv[0]]):
+        # geteuid is not root and
+        # the command was called with sudo
+        # and failed
+        sys.exit(error_message)
```

Comment:
@mgerstner this is based on https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/177#issuecomment-2194352748 .Can you please confirm my understanding.

A user that is not setup as a sudo user runs `cloudguestregistryauth`, we then detect the executing user not to be root and trigger `sudo cloudguestregistryauth` in a sub process. sudo in turn will find the rule for this script and allow execution in a root user context?

Is this correct?

Reply:
Mostly correct, yes, except for:

> A user that is not setup as a sudo user runs [...]

Any user can run the script. Only if the caller is not already root, then reexecution using `sudo` happens.

The sudoers rule you currently have allows _any_ user to run this script as root, therefore no special configuration is needed. As I outlined in the related Bugzilla bug, you could consider requiring membership in a dedicated group for hardening purposes. The sudoers rule would then need to look like this:

```
%cloudgroup ALL=(root) NOPASSWD: /usr/bin/cloudguestregistryauth ""
```

Otherwise the script and rule in this PR# look good to me now.